Source/DafnyCore/Options/BoogieOptionBag.cs

```diff
@@ -17,8 +19,34 @@ public static class BoogieOptionBag {
     ArgumentHelpName = "arguments",
   };
 
-  public static readonly Option<uint> Cores = new("--cores", () => 0,
-    "Run the Dafny CLI using <n> cores. Specifying 0 will use half of the available processor cores.") {
+  public static readonly Option<uint> Cores = new("--cores", result => {
+    if (result.Tokens.Count != 1) {
+      result.ErrorMessage = $"Expected only one value for option {Cores.Name}";
+      return 1;
+    }
+
+    var value = result.Tokens[0].Value;
+    if (value.EndsWith('%')) {
+      if (double.TryParse(value.Substring(0, value.Length - 1), out var percentage)) {
+        return Math.Max(1U, (uint)(percentage / 100.0 * Environment.ProcessorCount));
+      }
+
+      result.ErrorMessage = $"Could not parse percentage {value}";
+      return 1;
+    }
+
+    if (uint.TryParse(value, out var number)) {
+      if (number > 0) {
+        return number;
+      }
+
+      result.ErrorMessage = $"Number of cores to use must be greater than 0";
+      return 1;
+    }
+    result.ErrorMessage = $"Could not parse number {value}";
+    return 1;
+  }, true,
+    "Run the Dafny CLI using <n> cores, or using <XX%> of the machine's logical cores.") {
```

Comment:
```suggestion
    "Run the Dafny verifier using <n> cores, or using <XX%> of the machine's logical cores.") {
```
Is this text tested somewhere?

Reply:
I'm not sure I understand your question, the_text itself is not repeated somewhere else where it is diffed against.

Reply:
I mean, normally, when you change CLI options, the help at --help should reflect these changes. Aren't we testing the output of /help? Maybe we should in a future PR.